samples/Http2SampleApp/Program.cs

```diff
@@ -0,0 +1,50 @@
+using System.Globalization;
+using System.IO;
+using System.Net;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Http2SampleApp
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var configuration = new ConfigurationBuilder()
+                .AddEnvironmentVariables()
+                .Build();
+
+            if (!ushort.TryParse(configuration["BASE_PORT"], NumberStyles.None, CultureInfo.InvariantCulture, out var basePort))
+            {
+                basePort = 5000;
+            }
+
+            var hostBuilder = new WebHostBuilder()
+                .ConfigureLogging((_, factory) =>
+                {
+                    // Set logging to the MAX.
+                    factory.SetMinimumLevel(LogLevel.Trace);
+                    factory.AddConsole();
+                })
+                .UseKestrel(options =>
+                {
+                    // Run callbacks on the transport thread
+                    options.ApplicationSchedulingMode = SchedulingMode.Inline;
+
+                    options.Listen(IPAddress.Any, basePort, listenOptions =>
+                    {
+                        listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
```

Comment:
Could we just add this to our main sample app and use a command line argument to control whether or not to support HTTP/2?

Reply:
I'm hesitant to do that right now since the Sample app already has HTTPS configuration using PFX. Once we get the ALPN SslStream stuff merged we can look at merging the samples.

Reply:
That's fine. We could switch from UseHttps to UseTls in the main sample when HTTP/2 is selected, but no big deal either way.